### DISCUSSION TITLE
Apiato, the open source flawless framework for building scalable and testable API-Centric Apps with PHP and Laravel.

### DISCUSSION THREAD

**Discussion by mariano-pualiu** [HassanZahirnia/laravel-package-ocean#148] [Package Suggestions]
Features:

- Authentication with OAuth2.0 for first/third-party clients (using Laravel Passport).
- Role-Based Access Control (RBAC), seeded with a Super Admin, Roles and Permissions.
- Query Parameters support (orderBy, sortedBy and filter) with full-text search.
- Useful Endpoints for managing users, roles/permissions, tokens and more.
- API Documentations generator, to generate API docs from PHP Docblock using ApiDocJS (provided by Documentation Container).
- Supports CORS (Cross-Origin Resource Sharing) and JSONP (JSON with padding).
- Auto encoding/decoding of real ID's, to prevent exposing real ID's to the outer world.
- API Throttling (rate limiting to control the rate of traffic received and protect your server).
- Support "include" request parameter to load nested relationships.
- Support Data Caching (with auto clearing on Create, Update and Delete).
- API versioning in the URL or Header (versioning based on the route file name).
- Exception handling with custom error responses.
- Support Shallow ETag HTTP Header, to reduce client bandwidth.
- Localization - easily switch between languages via request header field (provided by Localization Container).
- Automatic Data Pagination (meta links to the next and previous data).
- WEB and API Auth Middlewares. With Proxy endpoints for safer authentication of first-party Apps.
- Http Requests/Response Monitor and DB Query Debugger (provided by Debugger Container).
- Profiler, to display profiling data in the response (using Laravel Debugbar).
- Social Authentication supported out of the box "Facebook, Twitter, Google+" Using Laravel Socialite (provided by SocialAuth Container).
- Type-Casting JSON responses and pagination results with Transformers (using Fractal).
- Useful Tests Helpers for faster and more enjoyable automated testing (using PHPUnit).
- Support multiple response payload formats (JSON API spec v1.0, Data Array & pure Data).
- Automatic dates conversion to ISO format in responses, from the Transformers.
- Support Stripe payment gateway, extendable to support other payment gateways (provided by Payment Container).
- System and user level settings out of the box (provided by Settings Container).
- Easy Validation and Authorization (for User data, accessibility and ownership) with Requests objects.
- Maintainable and scalable Software Architectural Pattern (using the Porto SAP).
- Code generator commands, for generating full CRUD operations, and other useful Commands.
- Separation of UI's (Web, API and CLI) components (routes, controllers, requests, tests...).
- Fully detailed documentation, with code samples, principles, definitions and guidelines.
- 100% customizable and open code. Using greatest frameworks, tools, packages and standards.

**Comment by HassanZahirnia** [comment]
Hey, I don't see any links.

**Comment by mariano-pualiu** [comment]
Sorry
https://apiato.io/
https://apiato.io/docs/
https://github.com/apiato/apiato

**Reply by me-shaon** [reply]
@mariano-pualiu Apiato is not a 'Laravel package'. Should it be listed here?
